packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -533,6 +533,14 @@ class _HeroFlight {
   }
 
   void _handleAnimationUpdate(AnimationStatus status) {
+    // The animation will not finish until the user lift their finger, so we
+    // should ignore the status update if the gesture is in progress.
+    //
+    // This also relies on the animation to update its status at the end of the
+    // gesture. See the _CupertinoBackGestureController.dragEnd for how
+    // cupertino page route achieves that.
+    if (manifest.fromRoute?.navigator?.userGestureInProgress == true)
```

Comment:
One thing that troubles me in this approach is that we push the responsibility, that gesture should trigger animation update when it ends, to the route transition class. I am worrying the third party library developer might trip on this

Reply:
I think we already crossed that bridge. Since Navigator and Hero are both from the widgets package and heroes.dart is already making use of the user gesture state.

Reply:
Does this still work if the user dragged it to the end and dropped it exactly there?

Reply:
yes we have tests to ensure that https://github.com/flutter/flutter/blob/fed18fc4331e6b109ff96be87cd209730b514246/packages/flutter/test/material/page_test.dart#L541
https://github.com/flutter/flutter/blob/fed18fc4331e6b109ff96be87cd209730b514246/packages/flutter/test/material/page_test.dart#L506